app/controller/AppController.php

```diff
@@ -65,25 +64,43 @@ public function accumulate(string $value)
     {
         try {
             $this->manager->append($value);
-            $chaine = $this->manager->getAccumulator();
-            $this->index($chaine);
-        } catch(Exception $e) {
+            $this->index();
+        } catch (Exception $e) {
             $this->error($e->getMessage());
         }
     }
 
     /**
      * Lance une action (opération, égal, pourcentage, ...)
      * @param string $action
+     * @throws Exception
      */
     public function action(string $action)
     {
-        switch($action) {
-            case 'clear';
-                $this->manager->reset();
-                break;
-            default;
+        try {
+            switch ($action) {
+                case 'clear':
+                    $this->manager->reset();
+                    break;
+                case 'plus':
+                case 'minus':
+                case 'times':
+                case 'divide':
+                    $this->manager->operator($action);
+                    break;
+                case 'equals':
+                    try {
```

Comment:
du coup si tu englobes ton switch par un `try/catch` tu ne dois pas mettre celui là. comme ça, toutes les exceptions lancées dans le try qui englobe le `switch` seront attrapées par le catch qui lui est associé. Ainsi, si une erreur est rencontrée, l'execution du try est stoppée et celle du catch est amorcée. Donc pas de `$this->index()` si il y a une erreur.

Reply:
Donc un seul try catch pour les 2 throw, soit 2 messages avec la même variable $e.